projects/cds/src/merchandising/cms-components/merchandising-carousel/merchandising-carousel.component.service.ts

```diff
@@ -70,31 +65,37 @@ export class MerchandisingCarouselComponentService {
     return metadata;
   }
 
-  private mapMerchandisingProductsToCarouselItems(
-    merchandisingProducts: MerchandisingProducts
+  private mapStrategyProductsToCarouselItems(
+    strategyProducts: StrategyProducts
   ): Observable<MerchandisingProduct>[] {
-    return merchandisingProducts && merchandisingProducts.products
-      ? merchandisingProducts.products.map((product, index) => {
-          product.metadata = this.getCarouselItemMetadata(product, index + 1);
-          return of(product);
-        })
+    return strategyProducts && strategyProducts.products
+      ? strategyProducts.products.map((strategyProduct, index) =>
+          this.productService.get(strategyProduct.id).pipe(
```

Comment:
https://github.com/SAP/cloud-commerce-spartacus-storefront/pull/5665/files#r354498491

the usage of strategyProduct